src/Microsoft.AspNetCore.Razor.Language/FileSystemRazorProject.cs

```diff
@@ -36,16 +39,24 @@ public override IEnumerable<RazorProjectItem> EnumerateItems(string basePath)
                 .EnumerateFiles("*.cshtml", SearchOption.AllDirectories)
                 .Select(file =>
                 {
-                    var relativePath = file.FullName.Substring(absoluteBasePath.Length).Replace(Path.DirectorySeparatorChar, '/');
-                    return new FileSystemRazorProjectItem(basePath, relativePath, file);
+                    var relativePhysicalPath = file.FullName.Substring(absoluteBasePath.Length + 1); // Include leading separator
+                    var filePath = "/" + relativePhysicalPath.Replace(Path.DirectorySeparatorChar, '/');
+
+                    return new FileSystemRazorProjectItem(basePath, filePath, relativePhysicalPath, file);
                 });
         }
 
         public override RazorProjectItem GetItem(string path)
         {
+            var absoluteBasePath = NormalizeAndEnsureValidPath("/");
             var absolutePath = NormalizeAndEnsureValidPath(path);
 
-            return new FileSystemRazorProjectItem("/", path, new FileInfo(absolutePath));
+            var file = new FileInfo(absolutePath);
+
+            var relativePhysicalPath = file.FullName.Substring(absoluteBasePath.Length + 1); // Include leading separator
+            var filePath = "/" + relativePhysicalPath.Replace(Path.DirectorySeparatorChar, '/');
+
+            return new FileSystemRazorProjectItem("/", filePath, relativePhysicalPath, new FileInfo(absolutePath));
```

Comment:
Rather than reuse `path` here, I'm computing `filePath` from the rest of the data, this way we're sure everything is correct.